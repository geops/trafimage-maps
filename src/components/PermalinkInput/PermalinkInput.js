import React from "react";
import { makeStyles } from "@mui/styles";
import UIPermalinkInput from "../PermalinkInputCore";

const useStyles = makeStyles(() => ({
  input: {
    display: "flex",
    justifyContent: "flex-end",
    width: "calc(100% - 4px)",
    height: 30,
    border: "1px solid lightgray",
    backgroundColor: "white",

    "& input,    & button,    & svg": {
      border: "none",
    },

    "& input": {
      flexGrow: 2,
      textOverflow: "ellipsis",
      minWidth: 0,
      paddingLeft: 5,

      "&::-ms-clear": {
        display: "none",
      },
    },

    "& button": {
      width: 30,
      height: "100%",
      flexShrink: 0,
      borderLeft: "1px solid lightgray",
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      padding: 7,
    },
  },
}));

function PermalinkInput(props) {
  const classes = useStyles();

  return <UIPermalinkInput className={classes.input} {...props} />;
}

export default React.memo(PermalinkInput);
