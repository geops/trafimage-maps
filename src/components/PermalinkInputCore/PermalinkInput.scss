.tm-permalink-field {
  display: flex;
  justify-content: flex-end;
  width: calc(100% - 4px);
  height: calc(100% - 4px);
  border: 1px solid $gray;
  background-color: white;

  input,
  button,
  svg {
    border: none;
  }

  input {
    flex-grow: 2;
    text-overflow: ellipsis;
    min-width: 0;

    &::-ms-clear {
      display: none;
    }
  }

  div[role='button'] {
    width: $btn-size-base;
    height: 100%;
    flex-shrink: 0;
  }
}
