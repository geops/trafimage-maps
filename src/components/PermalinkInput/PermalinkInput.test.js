import React from 'react';
import renderer from 'react-test-renderer';
import PermalinkInput from '.';

describe('PermalinkInput', () => {
  test('should match snapshot.', () => {
    const component = renderer.create(<PermalinkInput />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
