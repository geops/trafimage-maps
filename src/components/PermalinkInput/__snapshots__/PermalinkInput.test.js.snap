// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PermalinkInput should match snapshot. 1`] = `
<div class="makeStyles-input-1">
  <input type="text"
         tabindex="0"
         title
         class="tm-permalink-input"
         value
  >
  <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium tm-permalink-bt css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
          title
  >
    <svg stroke="currentColor"
         fill="currentColor"
         stroke-width="0"
         viewbox="0 0 24 24"
         focusable="false"
         height="1em"
         width="1em"
         xmlns="http://www.w3.org/2000/svg"
    >
      <path fill="none"
            d="M0 0h24v24H0z"
      >
      </path>
      <path d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z">
      </path>
    </svg>
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </button>
</div>
`;
